src/App.tsx

```diff
@@ -1,24 +1,125 @@
-import React, { useState } from 'react';
+import React, { useReducer } from 'react';
 import { StyleSheet, Text, View } from 'react-native';
 import AddItem from './components/AddItem';
 import AddPartyMember from './components/AddPartyMember';
 import Tax from './components/Tax';
 import Tip from './components/Tip';
 
+enum Actions {
+  ADD_ITEM = 'add_item',
+  ADD_PARTY_MEMBER = 'add_party_member',
+  SET_TIP = 'set_tip',
+  SET_TAX = 'set_tax',
+}
+
+type AppStateType = {
+  total: number;
+  items: {
+    [name: string]: {
+      price: number;
+      purchaserList: string[];
+    };
+  };
+  party: string[];
+  tip: number;
+  tax: number;
+};
+
+type ActionArgs = {
+  type: Actions;
+
+  // item name or party member name
+  name: string;
+
+  // item details
+  price: number;
```

Comment:
Because this attributes can be grouped, would it make sense to have different types to contain these details? So something like:
```
type ItemDetails = {
  price: number;
  purchaserList: string[];
}

enum AmountDetailsType {
  TIP = 'tip',
  TAX = 'tax'
}
type AmountDetails = {
  type: AmountDetailsType
  amount: number
}

type ActionArgs = {
  type: Actions;

  // item name or party member name
  name: string;

  itemDetails: ItemDetails;

  amountDetails: AmountDetails;
};
```

Reply:
Also, while writing that out, is ActionArgs being used for both item and party member name? :o If this is being used for part member name, the itemDetails and amountDetails are not filled right?

Reply:
fixed! i organized each one into its own object type details